File: programmers/42576.cpp
/*
 * 완주하지 못한 선수
 * https://programmers.co.kr/learn/courses/30/lessons/42576
 * sub_task - 100
 * efficiency - 0
 */

#include <string>
#include <vector>
#include <algorithm>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    for(int i=0; i<completion.size(); i++){
        vector<string>::iterator it = find(participant.begin(), participant.end(), completion[i]);
        if(it != participant.end()){
            participant.erase(it);
        }
    }
    answer = participant[0];
    return answer;
}

File: programmers/hash/42576.py
"""
완주하지 못한 선수
sub_task - 100
effiecincy - 100
dictionary, get(key, default_value) - if not key return default_value
"""

def solution(participant, completion):
    answer = {}
    for i in participant:
        answer[i] = answer.get(i, 0)+1
    for j in completion:
        answer[j] = answer[j]-1
    for k in answer:
        if answer[k]: return k
